test/testdata/infer/arg_matching.rb

```diff
@@ -54,8 +54,6 @@ def call_kwarg
     kwarg(1, b: "hi")
     #           ^^^^ error: Expected `Integer` but found `String("hi")` for argument `b`
     kwarg(1, any)
-    kwarg(1, a_hash)
-    #        ^^^^^^ error: Cannot call `TestArgs#kwarg` with a `Hash` keyword splat because the method has required keyword parameters
```

Comment:
Deleting all these errors is a regression, isn't it?

Sorbet previously rejected this code because it couldn't guarantee whether `a_hash` has one entry in it for every parameter that the `kwarg` method requires.

Reply:
I've deleted this test because:
1. It's a valid ruby 2.7 code which will work as expected
2. It's also a base case in the `test/testdata/infer/ruby3_keyword_args.rb`:
https://github.com/sorbet/sorbet/blob/d331f62b15b019c12b751e0c49ac2ae599e93999/test/testdata/infer/ruby3_keyword_args.rb#L6-L13

I've also tried getting that case back and running sorbet over it with `--experimental-ruby3-keyword-args`. This is what I got:
```
test/testdata/infer/arg_matching.rb:57: Keyword argument hash without ** is deprecated https://srb.help/7033
    57 |    kwarg(1, a_hash)
                     ^^^^^^
    test/testdata/infer/arg_matching.rb:57: This produces a runtime warning in Ruby 2.7, and will be an error in Ruby 3.0
    57 |    kwarg(1, a_hash)
                     ^^^^^^
  Autocorrect: Done
    test/testdata/infer/arg_matching.rb:57: Replaced with **a_hash
    57 |    kwarg(1, a_hash)
```

>Deleting all these errors is a regression, isn't it?

I don't believe deleting those errors is a regression. The detection we had before was too generic and also proposed to use `T.unsafe`. The new detection doesn't alert on valid code and proposes fixes without introducing unsafe code.